src/cmd/cli/command/commands.go

```diff
@@ -562,16 +562,37 @@ var whoamiCmd = &cobra.Command{
 		jsonMode, _ := cmd.Flags().GetBool("json")
 
 		token := cluster.GetExistingToken(getCluster())
-		userInfo, err := auth.FetchUserInfo(cmd.Context(), token)
-		if err != nil {
-			return err
+		var (
+			userInfo *auth.UserInfo
+			subject  string
+			issuer   string
+		)
+		if token != "" {
+			if claims, err := cli.ParseTokenClaims(token); err == nil {
+				subject = claims.Subject
+				issuer = claims.Issuer
+			} else {
+				term.Debug("unable to parse JWT claims:", err)
+			}
+		}
+
+		fabricToken := strings.Contains(strings.ToLower(issuer), "fabric")
+		if !fabricToken {
+			subject = "" // only show subject for Fabric tokens
```

Comment:
Let's get more specific than "fabric token" what is it about the token that makes it special in this case?